### PR TITLE
Use `to_string` instead of `format!`

### PR DESCRIPTION
**Pull request by TaKO8Ki**

### CODE REVIEW

**Comment by compiler-errors** [comment]
hm, @TaKO8Ki why did you change just this one and not other similar usages? anyways, I guess it's fine.

r? @compiler-errors 
@bors r+ rollup=always

**Comment by TaKO8Ki** [comment]
@compiler-errors I replaced rest of `format!("{}"..` with `to_string` in `compiler/`.

File: compiler/rustc_typeck/src/check/mod.rs [hunk 1]
```diff
@@ -749,7 +749,7 @@ fn fn_sig_suggestion<'tcx>(
             Some(match ty.kind() {
                 ty::Param(_) if assoc.fn_has_self_parameter && i == 0 => "self".to_string(),
                 ty::Ref(reg, ref_ty, mutability) if i == 0 => {
-                    let reg = match &format!("{}", reg)[..] {
```
Comment (compiler-errors) [inline compiler/rustc_typeck/src/check/mod.rs:752]: this is weird... can we do something to simplify the logic?

Comment (TaKO8Ki) [reply compiler/rustc_typeck/src/check/mod.rs:752]: like this?

```rust
&*reg.to_string()
```

Comment (compiler-errors) [reply compiler/rustc_typeck/src/check/mod.rs:752]: no, sorry -- i meant to point out the body of the match block. it can avoid calling to_string twice.

Comment (TaKO8Ki) [reply compiler/rustc_typeck/src/check/mod.rs]: I improved it to avoid calling `to_string` twice.

**Comment by compiler-errors** [comment]
@bors r+ rollup=always

please open another PR if you want to change the other usages of `format!("{}", ..)` instead of pushing to this branch after it's been approved. thanks!

**Pull request merged**